tests/xharness/Harness.cs

```diff
@@ -154,23 +156,18 @@ public class Harness : IHarness {
 		string MlaunchPath {
 			get {
 				if (ENABLE_DOTNET) {
-					string platform;
+					ApplePlatform platform;
 					if (INCLUDE_IOS) {
-						platform = "iOS";
+						platform = ApplePlatform.iOS;
 					} else if (INCLUDE_TVOS) {
-						platform = "tvOS";
+						platform = ApplePlatform.TVOS;
 					} else {
 						return $"Not building any mobile platform, so can't provide a location to mlaunch.";
 					}
-					var mlaunchPath = Path.Combine (DOTNET_DIR, "packs");
-					// there is a diff between getting the path for the current platform when running on CI or off CI. The config files in the CI do not 
-					// contain the correct workload version, the reason for this is that the workload is built in a different machine which means that
-					// the Make.config will use the wrong version. The CI set the version in the environment variable {platform}_WORKLOAD_VERSION via a script.
-					var workloadVersion = Environment.GetEnvironmentVariable ($"{platform.ToUpperInvariant ()}_WORKLOAD_VERSION");
-					mlaunchPath = Path.Combine (mlaunchPath, $"Microsoft.{platform}.Sdk",
-							string.IsNullOrEmpty (workloadVersion) ? config [$"{platform.ToUpperInvariant ()}_NUGET_VERSION_NO_METADATA"] : workloadVersion);
-					mlaunchPath = Path.Combine (mlaunchPath, "tools", "bin", "mlaunch");
-					return mlaunchPath;
+					var sdkPlatform = platform.AsString ().ToUpperInvariant ();
+					var sdkName = config [$"{sdkPlatform}_NUGET_SDK_NAME"];
+					var sdkVersion = config [$"{sdkPlatform}_NUGET_VERSION_NO_METADATA"];
+					return Path.Combine (DOTNET_DIR, "packs", sdkName, sdkVersion, "tools", "bin", "mlaunch");
```

Comment:
This is going to give use the wrong version in CI if I am not mistaken since we need to make sure that the version is the same as the one used in the machine that built the nugets, or is this not longer need?

Reply:
The version should be the same on all bots, if not, that's a bug somewhere else.

Reply:
@mandel-macaque fixed, please re-review!